### ISSUE TITLE
manual/scalaGuide/main/templates/ScalaCustomTemplateFormat.md

### ISSUE THREAD

**Issue by garbagetown** [playframework-ja/translation-project#1025]
- https://github.com/playframework-ja/translation-project/tree/2.3.x/manual/scalaGuide/main/templates/ScalaCustomTemplateFormat.md
- diffs between [2.2.0/manual/scalaGuide/main/templates/ScalaCustomTemplateFormat.md](https://github.com/playframework-ja/translation-project/tree/2.2.0/manual/scalaGuide/main/templates/ScalaCustomTemplateFormat.md) and [2.3.x/manual/scalaGuide/main/templates/ScalaCustomTemplateFormat.md](https://github.com/playframework-ja/translation-project/tree/2.3.x/manual/scalaGuide/main/templates/ScalaCustomTemplateFormat.md).

```
--- /Users/garbagetown/Desktop/2.2.0/manual/scalaGuide/main/templates/ScalaCustomTemplateFormat.md  2013-09-19 10:11:22.000000000 +0900
+++ //Users/garbagetown/Desktop/2.3.5/manual/scalaGuide/main/templates/ScalaCustomTemplateFormat.md 2014-07-28 12:26:38.000000000 +0900
@@ -1,8 +1,9 @@
+<!--- Copyright (C) 2009-2013 Typesafe Inc. <http://www.typesafe.com> -->
 # Adding support for a custom format to the template engine

 The built-in template engine supports common template formats (HTML, XML, etc.) but you can easily add support for your own formats, if needed. This page summarizes the steps to follow to support a custom format.

-## Overview of the the templating process
+## Overview of the templating process

 The template engine builds its result by appending static and dynamic content parts of a template. Consider for instance the following template:

@@ -24,13 +25,13 @@

 ## Implement a format

-Implement the `play.templates.Format[A]` trait that has the methods `raw(text: String): A` and `escape(text: String): A` that will be used to integrate static and dynamic template parts, respectively.
+Implement the `play.twirl.api.Format[A]` trait that has the methods `raw(text: String): A` and `escape(text: String): A` that will be used to integrate static and dynamic template parts, respectively.

-The type parameter `A` of the format defines the result type of the template rendering, e.g. `Html` for a HTML template. This type must be a subtype of the `play.templates.Appendable[A]` trait that defines how to concatenates parts together.
+The type parameter `A` of the format defines the result type of the template rendering, e.g. `Html` for a HTML template. This type must be a subtype of the `play.twirl.api.Appendable[A]` trait that defines how to concatenates parts together.

-For convenience, Play provides a `play.api.templates.BufferedContent[A]` abstract class that implements `play.templates.Appendable[A]` using a `StringBuilder` to build its result and that implements the `play.api.mvc.Content` trait so Play knows how to serialize it as an HTTP response body (see the last section of this page for details).
+For convenience, Play provides a `play.twirl.api.BufferedContent[A]` abstract class that implements `play.twirl.api.Appendable[A]` using a `StringBuilder` to build its result and that implements the `play.twirl.api.Content` trait so Play knows how to serialize it as an HTTP response body (see the last section of this page for details).

-In short, you need to write to classes: one defining the result (implementing `play.templates.Appendable[A]`) and one defining the text integration process (implementing `play.templates.Format[A]`). For instance, here is how the HTML format is defined:
+In short, you need to write to classes: one defining the result (implementing `play.twirl.api.Appendable[A]`) and one defining the text integration process (implementing `play.twirl.api.Format[A]`). For instance, here is how the HTML format is defined:

 '''scala
 // The `Html` result type. We extend `BufferedContent[Html]` rather than just `Appendable[Html]` so
@@ -47,13 +48,13 @@

 ## Associate a file extension to the format

-The templates are compiled into a `.scala` files by the build process just before compiling the whole application sources. The `sbt.PlayKeys.templatesTypes` key is a sbt setting of type `Map[String, String]` defining the mapping between file extensions and template formats. For instance, if HTML was not supported out of the box by Play, you would have to write the following in your build file to associate the `.scala.html` files to the `play.api.templates.HtmlFormat` format:
+The templates are compiled into a `.scala` files by the build process just before compiling the whole application sources. The `TwirlKeys.templateFormats` key is a sbt setting of type `Map[String, String]` defining the mapping between file extensions and template formats. For instance, if HTML was not supported out of the box by Play, you would have to write the following in your build file to associate the `.scala.html` files to the `play.twirl.api.HtmlFormat` format:

 '''scala
-templatesTypes += ("html" -> "play.api.templates.HtmlFormat")
+TwirlKeys.templateFormats += ("html" -> "my.HtmlFormat.instance")
 '''

-Note that the right side of the arrow contains the fully qualified name of a value of type `play.templates.Format[_]`.
+Note that the right side of the arrow contains the fully qualified name of a value of type `play.twirl.api.Format[_]`.

 ## Tell Play how to make an HTTP result from a template result type

@@ -64,7 +65,7 @@
   Writeable[Http](result => codec.encode(result.body), Some(ContentTypes.HTTP))
 '''

-> **Note:** if your template result type extends `play.api.templates.BufferedContent` you only need to define an
+> **Note:** if your template result type extends `play.twirl.api.BufferedContent` you only need to define an
 > implicit `play.api.http.ContentTypeOf` value:
 > '''scala
 > implicit def contentTypeHttp(implicit codec: Codec): ContentTypeOf[Http] =

```

**Comment by garbagetown** [comment]
:hand:

**Comment by garbagetown** [comment]
merged at #1100
